jupyter_client/kernelspec.py

```diff
@@ -242,10 +273,12 @@ def install_kernel_spec(self, source_dir, kernel_name=None, user=False,
         if not kernel_name:
             kernel_name = os.path.basename(source_dir)
         kernel_name = kernel_name.lower()
-        
+        if not _valid_kernel_name(kernel_name):
+            raise ValueError("Invalid kernel name %r.  %s" % (kernel_name, _kernel_name_description))
```

Comment:
There are two places right now:

1. installing new specs (I'd like to raise here)
2. listing existing specs (we can reduce this to a warning, to avoid disabling already-installed specs)